### PR TITLE
cgroup: Fix python3 str type issue

### PR DESCRIPTION
**Pull request by xiangchunfu**
decode from bytes to str for python3

id: 1745847

Signed-off-by: Xiangchun Fu xfu@redhat.com

### CODE REVIEW

File: qemu/tests/cgroup.py [hunk 1]
```diff
@@ -195,6 +195,7 @@ def param_add_scsi_disks():
                            "add_host", shell=True)
             time.sleep(1)   # Wait for device init
             dev = process.system_output("ls /dev/sd* | tail -n 1", shell=True)
```
Comment (PaulYuuu) [inline qemu/tests/cgroup.py]:
```suggestion
             dev = process.run("ls /dev/sd* | tail -n 1", shell=True).stdout_text
```

Comment (xiangchunfu) [reply qemu/tests/cgroup.py]: @PaulYuuu  Thanks for your review.  seems It doesn't works.  This is test log.
05:38:57 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/cgroup.py", line 199, in param_add_scsi_disks
    dev = dev.decode()
05:38:57 ERROR| AttributeError: 'str' object has no attribute 'decode'

If comment   #dev = dev.decode(). 
...
stderr: b'/bin/sh: /sys/block/sdc: Is a directory\n/bin/sh: line 1: /queue/rotational: No such file or directory\n'...

Comment (PaulYuuu) [reply qemu/tests/cgroup.py]: @xiangchunfu I mean remove the new code and only need to modify the old one.
https://github.com/avocado-framework/avocado/blob/bdf5daad93e8a4f9e542195c7c6ac222d7d32918/avocado/utils/process.py#L350-L355

Comment (PaulYuuu) [reply qemu/tests/cgroup.py]: > If comment #dev = dev.decode().
> ...
> stderr: b'/bin/sh: /sys/block/sdc: Is a directory\n/bin/sh: line 1: /queue/rotational: No such file or directory\n'...

Looks like `stdout_text` will include `\n`, so `process.getoutput("ls /dev/sd* | tail -n 1", shell=True)` or `process.run("ls /dev/sd* | tail -n 1", shell=True).stdout_text.strip()` can fix it.

Comment (xiangchunfu) [reply qemu/tests/cgroup.py]: @PaulYuuu  Thanks.  It works.  updated

**Comment by PaulYuuu** [comment]
Ack

**Comment by xiangchunfu** [comment]
(1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.cgroup.blkio_bandwidth: PASS (409.21 s)

**Comment by xiangchunfu** [comment]
@vivianQizhu  could you help review this, thanks!

File: qemu/tests/cgroup.py [hunk 2]
```diff
@@ -194,7 +194,7 @@ def param_add_scsi_disks():
             process.system("echo 1 > /sys/bus/pseudo/drivers/scsi_debug/"
                            "add_host", shell=True)
             time.sleep(1)   # Wait for device init
-            dev = process.system_output("ls /dev/sd* | tail -n 1", shell=True)
+            dev = process.run("ls /dev/sd* | tail -n 1", shell=True).stdout_text.strip()
```
Comment (vivianQizhu) [inline qemu/tests/cgroup.py]: `process.getoutput()`

File: qemu/tests/cfg/cgroup.cfg [hunk 3]
```diff
@@ -14,6 +14,11 @@
             take_regular_screendumps = "no"
             image_snapshot = yes
             cgroup_test = "blkio_bandwidth"
+            Host_RHEL.m8:
+                blkio.weight.file = "blkio.bfq.weight"
```
Comment (vivianQizhu) [inline qemu/tests/cfg/cgroup.cfg]: Better use `_` replaces `.`

**Comment by xiangchunfu** [comment]
@vivianQizhu Thanks, Upated.  It works.

 (1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.cgroup.blkio_bandwidth: PASS (409.64 s)

File: qemu/tests/cgroup.py [hunk 4]
```diff
@@ -357,7 +357,8 @@ def _test(direction):
         for i in range(no_vms):
             blkio.mk_cgroup()
             assign_vm_into_cgroup(vms[i], blkio, i)
-            blkio.set_property("blkio.weight", weights[i], i)
+            blkio.weight = params.get("blkio_weight_file")
```
Comment (vivianQizhu) [inline qemu/tests/cgroup.py]: `blkio_weight` if only a variant.

File: qemu/tests/cgroup.py [hunk 5]
```diff
@@ -194,7 +194,7 @@ def param_add_scsi_disks():
             process.system("echo 1 > /sys/bus/pseudo/drivers/scsi_debug/"
                            "add_host", shell=True)
             time.sleep(1)   # Wait for device init
-            dev = process.system_output("ls /dev/sd* | tail -n 1", shell=True)
+            dev = process.getoutput("ls /dev/sd* | tail -n 1", shell=True).stdout_text.strip()
```
Comment (vivianQizhu) [inline qemu/tests/cgroup.py]: `.stdout_text.strip()` in not necessary here.

File: qemu/tests/cgroup.py [hunk 6]
```diff
@@ -1684,8 +1685,8 @@ def _get_stat(pid):
             out = None
             for i in range(10):
                 try:
-                    out = process.system_output("cat /proc/%s/task/*/stat" %
-                                                pid, shell=True)
+                    out = process.getoutput("cat /proc/%s/task/*/stat" %
+                                            pid, shell=True).stdout_text.strip()
```
Comment (vivianQizhu) [inline qemu/tests/cgroup.py]: Same here.

**Comment by xiangchunfu** [comment]
(1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.cgroup.blkio_bandwidth: PASS (411.11 s)

File: qemu/tests/cgroup.py (hunk 4 above)
Comment (xiangchunfu) [reply qemu/tests/cgroup.py]: updated

File: qemu/tests/cgroup.py (hunk 5 above)
Comment (xiangchunfu) [reply qemu/tests/cgroup.py]: updated

**Comment by vivianQizhu** [comment]
@xiangchunfu Please fix CI failure.

File: qemu/tests/cgroup.py [hunk 7]
```diff
@@ -1684,8 +1685,7 @@ def _get_stat(pid):
             out = None
             for i in range(10):
                 try:
-                    out = process.system_output("cat /proc/%s/task/*/stat" %
-                                                pid, shell=True)
+                    out = process.getoutput("cat /proc/%s/task/*/stat" % pid, shell=True)
```
Comment (vivianQizhu) [inline qemu/tests/cgroup.py]: Keep lines within 79 characters, thanks.

Comment (xiangchunfu) [reply qemu/tests/cgroup.py]: updated, thanks

**Pull request merged**